src/components/TldrawCanvas.tsx

```diff
@@ -944,107 +1079,196 @@ const TldrawCanvas = ({ title }: Props) => {
-                          discourseContext.nodes[relation.source].text;
-                        if (source.type !== relation.source) {
-                          return deleteAndWarn(
-                            `Source node must be of type ${sourceLabel}`
-                          );
-                        }
-                        const possibleTargets = discourseContext.relations[
-                          relation.label
-                        ]
-                          .filter((r) => r.source === relation.source)
-                          .map((r) => r.destination);
-                        if (!possibleTargets.includes(target.type)) {
-                          return deleteAndWarn(
-                            `Target node must be of type ${possibleTargets
-                              .map((t) => discourseContext.nodes[t].text)
-                              .join(", ")}`
-                          );
-                        }
-                        if (arrow.type !== target.type) {
+
+                        // Handle "Add Referenced Node" Arrows
+                        if (
+                          new Set(allAddReferencedNodeActions).has(arrow.type)
+                        ) {
+                          const possibleTargets = allAddReferencedNodeByAction[
+                            arrow.type
+                          ].map((action) => action.destinationType);
+                          if (!possibleTargets.includes(target.type)) {
+                            return deleteAndWarn(
+                              `Target node must be of type ${possibleTargets
+                                .map((t) => discourseContext.nodes[t].text)
+                                .join(", ")}`
+                            );
+                          }
+
+                          // source and target are expected to be pages
+                          // TODO: support blocks
+                          const targetTitle = target.props.title;
+                          const sourceTitle = source.props.title;
+                          const isTargetTitleCurrent =
+                            getPageTitleByPageUid(target.props.uid).trim() ===
+                            targetTitle.trim();
+                          const isSourceTitleCurrent =
+                            getPageTitleByPageUid(source.props.uid).trim() ===
+                            sourceTitle.trim();
+                          if (!isTargetTitleCurrent || !isSourceTitleCurrent) {
+                            return deleteAndWarn(
+                              "Either the source or target node has been renamed. Please update the nodes and try again."
+                            );
+                          }
+
+                          // Hack for default shipped EVD format: [[EVD]] - {content} - {Source},
+                          // replace when migrating from format to specification
+                          let newTitle: string;
```

Comment:
```typescript
const format = "[[EVD]] - {content} - {Source}";
const newTitle = format.replace("{content}", targetTitle).replace("{Source}", `[[${sourceTitle}]]`)
```

Reply:
Ah.  `targetTitle` wouldn't just be `{content}`, it would contain the `[[EVD]] - `, so the output would be `[[EVD]] - [[EVD]] - `

I'm not sure how to grab `{content}` based on the arbitrary format string.  I believe this needs to be solved with specification.

Maybe we could create a dynamic regex pattern to capture `{content}`? I imagine it would be similar to how format is converted to `has title` in specification currently.